packages/integration/lib/src/widgets/post/comment/comment_list_widget.dart

```diff
@@ -0,0 +1,517 @@
+        eventName: LMFeedAnalyticsKeys.commentListOpen,
+        widgetSource: LMFeedWidgetSource.postDetailScreen,
+        eventProperties: {
+          'postId': widget.postId,
+        },
+      ),
+    );
+    _addPaginationListener();
+  }
+
+  @override
+  void didUpdateWidget(covariant LMFeedCommentList oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    _addPaginationListener();
+  }
+
+  void _addPaginationListener() {
+    _commentListPagingController.addPageRequestListener(
+      (pageKey) {
+        _commentBloc.add(
+          LMFeedGetCommentsEvent(
+            postId: widget.postId,
+            page: pageKey,
+            pageSize: _pageSize,
+          ),
+        );
+      },
+    );
+  }
+
+  // This function updates the paging controller based on the state changes
+  void updatePagingControllers(LMFeedCommentState state) {
+    if (state is LMFeedGetCommentSuccessState) {
+      _postViewData = state.post;
+      final isLastPage = state.comments.length < _pageSize;
+      if (isLastPage) {
+        _commentListPagingController.appendLastPage(state.comments);
+      } else {
+        _commentListPagingController.appendPage(state.comments, state.page + 1);
+      }
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocConsumer<LMFeedCommentBloc, LMFeedCommentState>(
+        bloc: _commentBloc,
+        listener: _handleBlocListeners,
+        buildWhen: (previous, current) {
+          if (current is LMFeedAddCommentSuccessState ||
+              current is LMFeedAddCommentErrorState ||
+              current is LMFeedEditCommentSuccessState ||
+              current is LMFeedEditCommentErrorState ||
+              current is LMFeedDeleteCommentSuccessState ||
+              current is LMFeedDeleteCommentErrorState ||
+              current is LMFeedReplyCommentSuccessState ||
+              current is LMFeedDeleteReplySuccessState ||
+              current is LMFeedGetCommentSuccessState ||
+              current is LMFeedGetReplyCommentLoadingState ||
+              current is LMFeedCloseReplyState ||
+              current is LMFeedCommentRefreshState) {
+            return true;
+          }
+          return false;
+        },
+        builder: (context, state) {
+          return PagedSliverList.separated(
+            pagingController: _commentListPagingController,
+            builderDelegate: PagedChildBuilderDelegate<LMCommentViewData>(
+              firstPageProgressIndicatorBuilder: (context) {
+                return const Padding(
+                  padding: EdgeInsets.only(top: 150.0),
+                  child: LMFeedLoader(),
+                );
+              },
+              newPageProgressIndicatorBuilder: (context) {
+                return _widgetBuilder
+                    .newPageProgressIndicatorBuilderFeed(context);
+              },
+              noItemsFoundIndicatorBuilder: (context) =>
+                  const LMFeedEmptyCommentWidget(),
```

Comment:
Where is this widget declared? We should ideally not create new widgets for these cases. Plus missing customisation.

Reply:
It was built already under `packages/integration/lib/src/widgets/post/comment/default_empty_comment_widget.dart`, and previously used in post detail screen as well. lmk if i need to remove it

Reply:
Just take care of customisation

Reply:
done